src/adapter/src/catalog/storage.rs

```diff
@@ -1558,13 +1558,11 @@ impl<'a> Transaction<'a> {
     }
 
     /// Upserts persisted system configuration `name` to `value`.
-    pub fn upsert_system_config(&mut self, name: &str, value: &str) -> Result<(), Error> {
+    pub fn upsert_system_config(&mut self, name: &str, value: String) -> Result<(), Error> {
         let key = ServerConfigurationKey {
             name: name.to_string(),
         };
-        let value = ServerConfigurationValue {
-            value: value.to_string(),
-        };
+        let value = ServerConfigurationValue { value };
```

Comment:
Another thing to consider would be to change `ServerConfigurationValue.name` to be `Vec<String>` instead of `String`. That way you'd never need to unflatten the values and your persisted values would more closely resemble what was actually passed to `SystemVars::set`.

I haven't thought too hard if that makes sense though and if it's even possible with migrations. Just thought I'd float the idea.

Reply:
This was exactly my first implementation, but it fought too much with SynchronizedParameters and I think this is (for now) the better choice. We could still move to that in the future, but I don't think it's necessary for now. I've worked on this long enough I want to get the minimum improvement in first.

Reply:
I actually think that strings are the right long term design too, as that matches PostgreSQL. Variables are always serialized as a flat string, both in the catalog and in LaunchDarkly. They're only deserialized into native Rust types when you load them into a `SystemVars` or a `SessionVars`. Variables can be bools or floats or ints, and we're not trying to serialize _those_ as anything besides strings—so why would we treat string lists and ident lists specially? Lists are just another data type that can be represented easily as a flat string!

The fact that `SET` has dedicated syntax for lists is a weird special case—and I think this whole mess is much clearer if you think of `SET` as a weird special case.